### ISSUE TITLE
Model fade in at start of battle doesn't happen

### ISSUE THREAD

**Issue by Trivial-Man** [Albeoris/Memoria#223]
In the PSX version of the game enemies would sometimes fade in at the start of battle instead of being present right from the start. It may have been a simple technical limitation, but this gave the impression of them sort of showing up for battle if that makes sense. Similarly, while the battle camera was moving around showing off the enemies the player characters weren't present. They would also fade in after that was all done to emphasize the enemies. This isn't present at all in the remake and adding it back would be a nice enhancement

**Comment by Tirlititi** [comment]
Added in c5680fa.
Hopefully the timing is good. The effect looks similar to the PSX version for my tests, but maybe it will be a bit weird on some battles.

**Comment by Trivial-Man** [comment]
Goodness, I am sorry for the tardiness on this response. I see that you addressed a number of the issues I submitted and in general I'm very happy with all of them. This one as well is certainly much more similar to the PS1 release, however I should note that it could still be improved. In the implementation the mod is using currently the enemies and the player characters all load in at the same time. In the PS1 version the enemies would usually load in quite quickly and be the only thing that was visible while the camera moved about during the pre-battle. The player characters would usually only load in once the battle camera was nearly done moving and shortly before the status display loaded in indicating the battle clock started ticking. However this isn't always the case! I suspect all random encounters behave more or less the same, but for certain bosses or scripted encounters occasionally the player characters load in well before the camera stops moving. The solo fight with Zidane vs Baku in the crashed Prima Vista is the clearest example of this where Zidane and Baku seem to load in simultaneously, but it can also be seen in the 3rd battle against Beatrix in Alexandria and even the battle against Ozma. I am not certain if it is a consistent amount of time that the delay occurs for or if it is defined on a per-battle basis. It may even be that it is loading them in as fast as possible and the camera movements are an attempt to mask that they fade in, though in some fights such as against Gizamaluke it is very prominently visible so I somewhat doubt that.

The tl;dr of this is that while I think this is closer to the original release, only half of my initial concern was addressed. Adding a delay to displaying the player characters after the enemies load would get things a lot closer. But I suspect that to nail it perfectly it would have to be integrated into the thus far not understood camera movement system so that fading in of each side of the battle is only triggered once certain camera movements take place.
